common/c/targets/esp8266/target.c

```diff
@@ -24,104 +24,119 @@
-    if (!wifi_connect(SSID,PWD)) {
-        printf("Couldn't connect to \"%s\" with password \"%s\"\r\n2",
-               SSID, PWD);
-        goto loop;
+static void main_task(void *pvParameters);
+
+static int wifi_init(void);
+static int wifi_connect(const char *ssid, const char *pwd);
+static void wifi_print_opmode(void);
+static void wifi_print_station_config(void);
+
+static void uart_init(void);
+
+int target_initialize(void) {
+    if (wifi_init()) {
+        printf("Error initialising wifi!\r\n");
+        return 1;
     }
-        int ret = main();
-        printf("main() exited, returned %d\r\n", ret);
-        loop:
-            while(1);
+
+    if (wifi_connect(WIFI_SSID,WIFI_PASSWORD)) {
+        printf("Couldn't connect to \"%s\" with password \"%s\"\r\n",WIFI_SSID, WIFI_PASSWORD);
+        return 1;
+    }
+
+    return 0;
 }
 
 void ICACHE_FLASH_ATTR user_init() {
     uart_init();
-    if (!wifi_init()) {
-        printf("Error initialising wifi!\r\n");
-        while(1);
+
+    portBASE_TYPE error = xTaskCreate(main_task, (const signed char *)"main_task",
+                                      MAIN_STACK_SIZE, NULL, 2, NULL );
+    if (error<0) {
+        printf("Error creating main_task! Error code: %ld\r\n", error);
     }
-        portBASE_TYPE error = xTaskCreate(main_task, (const signed char *)"main_task",
-                                          MAIN_STACK_SIZE, NULL, 2, NULL );
-        if (error<0)
-            printf("Error creating main_task! Error code: %ld\r\n", error);
 }
 
-void uart_init() {
+static void main_task(void *pvParameters) {
+    (void)pvParameters;
+    printf("main_task() started\r\n");
+    int ret = main();
+    printf("main() exited with %d\r\n", ret);
+    for(;;);
+}
+
+
+static void uart_init(void) {
     uart_init_new();
     UART_SetBaudrate(UART0, 115200);
     UART_SetPrintPort(UART0);
 }
 
-bool wifi_init() {
+static int wifi_init(void) {
     printf("\r\nInitialising wifi station\r\n");
     wifi_print_opmode();
 
     if (!wifi_set_opmode_current(0x01)) {
         printf("Error setting wifi opmode to station mode!\r\n");
-        return false;
+        return 1;
     }
         printf("Changed wifi mode to station\r\n");
-        return true;
+        return 0;
 }
 
-bool wifi_connect(const char *ssid, const char *pwd) {
+static int wifi_connect(const char *ssid, const char *pwd) {
     struct station_config sta_cfg;
     memset(&sta_cfg, 0, sizeof(sta_cfg));
     strcpy((char *)sta_cfg.ssid, ssid);
     strcpy((char *)sta_cfg.password, pwd);
     if (!wifi_station_set_config_current(&sta_cfg)) {
         printf("Error setting wifi station config!\r\n");
-        return false;
+        return 1;
+    }
+
+    wifi_print_station_config();
+    printf("Connecting to %s...\r\n ", ssid);
+    if (!wifi_station_connect()) {
+        printf("FAIL!\r\n");
+        return 0;
+    }
+
+    uint8 status;
+    while((status = wifi_station_get_connect_status()) == STATION_CONNECTING);
```

Comment:
Uninline `(status = wifi_station_get_connect_status())`. We all love K&R but that's a bad style.